fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/client/item/CustomTexturedArmorRegistryImpl.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.client.item;
+
+import java.util.Map;
+
+import it.unimi.dsi.fastutil.objects.Reference2ObjectOpenHashMap;
+
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.api.client.item.v1.CustomTexturedArmorProvider;
+import net.fabricmc.fabric.api.client.item.v1.CustomTexturedArmorRegistry;
+
+public final class CustomTexturedArmorRegistryImpl implements CustomTexturedArmorRegistry {
+	public static final CustomTexturedArmorRegistry INSTANCE = new CustomTexturedArmorRegistryImpl();
+	private final Map<Item, CustomTexturedArmorProvider> providerMap = new Reference2ObjectOpenHashMap<>();
+
+	private CustomTexturedArmorRegistryImpl() {
+	}
+
+	@Override
+	public void registerCustomArmorTexture(CustomTexturedArmorProvider provider, Item... items) {
+		for (Item item : items) {
+			providerMap.put(item, provider);
```

Comment:
What about duplicates, how should we handle those?

Reply:
we don't, the one registered last gets it

Reply:
That somewhat seems like a load order battle

Reply:
you can't have two textures at the same time, also that's rare since usually you only edit your own item. If two mods edit the same vanilla item, you can't guarantee compatibility between them at all.